### PR TITLE
Expand GeoJSON filetype detection logic

### PR DESCRIPTION
**Pull request by who8mycakes**
Resolving GeoJSON detection issues raised here:
*https://github.com/mapbox/mapbox-file-sniff/issues/44 
*https://github.com/mapbox/mapbox-file-sniff/issues/35

Sniffing for `crs` and `features` as top-level parameters before looking for `type` expands the detection for GeoJSON file types.

### CODE REVIEW

**Comment by mapsam** [comment]
> smelloscope

😆

File: index.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ function sniff(buffer, callback) {
 
         if (header.indexOf('\"tilejson\":') !== -1) return callback(null, 'tilejson');
         if ((header.indexOf('\"arcs\":') !== -1) || (header.indexOf('\"objects\":') !== -1)) return callback(null, 'topojson');
+        if ((header.indexOf('\"crs\":') !== -1) || (header.indexOf('\"features\":') !== -1)) return callback(null, 'geojson');
```
Comment (mapsam) [inline index.js]: Just a sanity check for myself, this says "if `crs` OR `features` exists in the string, pass it on as a geojson"? Just taking a look at the TopoJSON specification, which does not specify `crs` anywhere, but _it could_ be added by a user if they wanted: https://github.com/mbostock/topojson-specification/issues/5 - I'm not worried about this use case, though.

File: index.js [hunk 2]
```diff
@@ -38,9 +39,7 @@ function sniff(buffer, callback) {
                 m[1] === 'Polygon' ||
                 m[1] === 'MultiPolygon' ||
                 m[1] === 'GeometryCollection') return callback(null, 'geojson');
-        }
-
-        return callback(invalid('Unknown filetype'));
+        }return callback(invalid('Unknown filetype'));
```
Comment (mapsam) [inline index.js]: could we put the `return` on a separate line?

**Comment by mapsam** [comment]
Looking good @who8mycakes! Just a little nitpicky syntax fix I mentioned for the `return` statement and this is great. Probably good tagging as `0.5.2`.

**Comment by sgillies** [comment]
@who8mycakes looks good to me.

**Comment by GretaCB** [comment]
Staging looks 👍 . Merging!

**Pull request merged**